gdal/frmts/saga/sagadataset.cpp

```diff
@@ -406,16 +406,45 @@ GDALDataset *SAGADataset::Open( GDALOpenInfo * poOpenInfo )
 
 {
     /* -------------------------------------------------------------------- */
-    /*  We assume the user is pointing to the binary (i.e. .sdat) file.     */
+    /*  We assume the user is pointing to the binary (i.e. .sdat) file or a */
+    /*  compressed raster (.sg-grd-z) file.                                 */
     /* -------------------------------------------------------------------- */
-    if( !EQUAL(CPLGetExtension( poOpenInfo->pszFilename ), "sdat"))
+    if( !EQUAL(CPLGetExtension( poOpenInfo->pszFilename ), "sdat") &&
```

Comment:
use a CPLString osExtension(CPLGetExtension( poOpenInfo->pszFilename )) variable to avoid code duplication